src/test/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionServiceTest.java

```diff
@@ -133,6 +133,87 @@ public void cleanupWhenDeletionFailsDoesntThrowException() {
     verify(bucket).delete();
   }
 
+  @Test
+  public void cleanupWhenDeletionThrowsExceptionOnBlobDoesntThrowException() {
+    BigQueryExtractionData data = new BigQueryExtractionData();
+    TableId tableId = TableId.of("dataset", "table");
+    when(table.getTableId()).thenReturn(tableId);
+    when(storage.delete(any(BlobId.class))).thenThrow(new RuntimeException());
+    Bucket bucket = mock(Bucket.class);
+    when(bucket.delete()).thenReturn(true);
+    when(storage.get(anyString())).thenReturn(bucket);
+    Blob blob = mock(Blob.class);
+    List<Blob> blobs = ImmutableList.of(blob);
+    Page pages = mock(Page.class);
+    when(storage.list(anyString())).thenReturn(pages);
+    when(pages.iterateAll()).thenReturn(blobs);
+
+    service.cleanup(data);
+    verify(storage).delete(any(BlobId.class));
+    verify(bucket).delete();
+  }
+
+  @Test
+  public void cleanupWhenDeletionThrowsExceptionOnBlobDoesntStillCleansRemainingBlobs() {
+    BigQueryExtractionData data = new BigQueryExtractionData();
+    TableId tableId = TableId.of("dataset", "table");
+    when(table.getTableId()).thenReturn(tableId);
+    when(storage.delete(any(BlobId.class))).thenReturn(true);
+    Bucket bucket = mock(Bucket.class);
+    when(bucket.delete()).thenReturn(true);
+    when(storage.get(anyString())).thenReturn(bucket);
+
+    List<Blob> blobs = new ArrayList<>();
```

Comment:
You could make this simpler and just have 3 blobs and throw an exception on the second call to delete() - mockito lets you set up mutliple actions on calls to a mocked method to do this.